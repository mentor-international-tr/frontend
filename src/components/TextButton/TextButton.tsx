import React, { isValidElement } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";

const styles = (theme: any) => ({
  button: {
    margin: theme.spacing.unit,
  },
  input: {
    display: "none",
  },
});

export interface TextButtonProps {
  classes: any;
  children: React.ReactElement | string;
  btnProps?: object;
}

const TextButton: React.SFC<TextButtonProps> = ({
  classes,
  children,
  btnProps,
}) => {
  return (
    <Button className={classes.button} {...btnProps}>
      {isValidElement(children)
        ? React.cloneElement(children)
        : children}
    </Button>
  );
};
TextButton.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(TextButton);
