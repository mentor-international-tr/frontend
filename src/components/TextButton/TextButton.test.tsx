import React from "react";
import { render, cleanup } from "react-testing-library";
import TextButton from "./TextButton";
import "jest-dom/extend-expect";

afterEach(cleanup);

describe("testing <TextButton /> component", () => {
  it("creates snapshot", () => {
    const { asFragment, rerender } = render(
      <TextButton btnProps={{
          color: "primary"
      }}>Message</TextButton>,
    );
    expect(asFragment()).toMatchSnapshot();
    })
})
