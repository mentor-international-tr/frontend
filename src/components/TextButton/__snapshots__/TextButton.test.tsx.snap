// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <TextButton /> component creates snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-textPrimary-6 MuiButton-flat-8 MuiButton-flatPrimary-9 TextButton-button-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-4"
    >
      Message
    </span>
    <span
      class="MuiTouchRipple-root-32"
    />
  </button>
</DocumentFragment>
`;
